src/screens/Customers/EditCustomer.tsx

```diff
@@ -15,28 +25,106 @@ const styles = (theme: Theme) =>
+      id: '',
+      dateUpdated: formatUTCDateStringToLocal((new Date()).toString()),
+      customerIds: [currentCustomer.id],
+      explanation: reason,
+      userId: user.id,
+    };
+
+    // Make a deep copy of an empty customer record
+    const customer: CustomerRecord = JSON.parse(JSON.stringify(currentCustomer));
+    customer.name = customerName;
+    customer.hasmeter = hasMeter;
+    customer.isactive = !customerInactive;
+    customer.tariffPlanId = selectedTariffPlanId;
+
+    if (customer.customerUpdates === undefined) {
+      customer.customerUpdates = [customerUpdate];
+    } else {
+      customer.customerUpdates.push(customerUpdate);
+    }
+
+    // TODO: add error handling
+    editCustomerInRedux(customer);
+    editCustomer({
+      ...customer,
+      customerUpdate: customerUpdate
+    });
+
+    // Navigate to the new customer's profile page
+    history.replace(`/customers/customer`, { customer });
+  }
+
+  const handleNameInput = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setCustomerName(event.target.value as string);
+  }
+
+  const handleReasonInput = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setReason(event.target.value as string);
+  }
+
+  const tariffPlans = currentSite.tariffPlans;
 
   return (
     <BaseScreen title="Edit Customer" leftIcon="backNav">
       <form noValidate className={classes.content}>
-        <TextField label={'Name'} id={'name'}/>
-        <Checkbox label={'Select if customer is inactive'} />
-        <Checkbox label={'Select if customer has meter'} />
-        <TextField label={'Tariff Plan'} id={'tariff-plan'}/>
-        <TextField label={'Reason'} id={'reason'}/>
-        <Button label={'SAVE'} />
+        <TextField label={'Name'} id={'name'} value={customerName} onChange={handleNameInput} />
+        <Checkbox label={'Select if customer is inactive'} checked={customerInactive} checkboxOnChange = {() => setCustomerInactive(!customerInactive)} />
+        <Checkbox label={'Select if customer has meter'} checked={hasMeter} checkboxOnChange={() => setHasMeter(!hasMeter)} />
+        <FormControl variant="outlined" className={classes.formControl}>
+          <InputLabel id="select-tariff-plan-label">Select Tariff Plan</InputLabel>
+          <Select label={"Select Tariff Plan"} id={'select-tariff-plan'} labelId = "select-tariff-plan-label" value={selectedTariffPlanId} onChange={handleSelectTariffPlan}>
+            {tariffPlans.map((plan) =>
+                <MenuItem key={plan.id} value={plan.id}>{plan.name}</MenuItem>
+            )}
+          </Select>
+        </FormControl>
+        <TextField label={'Reason'} id={'reason'} onChange={handleReasonInput}/>
+        <Button label={'SAVE'} onClick={handleSubmit} />
       </form>
     </BaseScreen>
   );
 }
 
-export default withStyles(styles)(EditCustomer);
+const mapStateToProps = (state: RootState) => ({
+  currentCustomer: state.siteData.currentCustomer || EMPTY_CUSTOMER,
```

Comment:
I'm confused why `currentCustomer` is part of `siteData` at all. Why is this necessary/why can't it just be passed around where needed? Also because it's not being reset/cleared when you  navigate away from the customer profile.

Reply:
Ok yeah I just saw Julian's note on this (https://github.com/calblueprint/meepanyar/pull/55/files#r598159199)

Reply:
Actually relying on `currentCustomer` instead of props for this screen could resolve your issue of having to go back twice (I spent a while digging into react router trying to come up with a fix for that but I couldn't find much).